docs/tutorials/cloud-init.md

```diff
@@ -0,0 +1,195 @@
+# Automatic setup with cloud-init
+*Authored by Carlos Nihelton ([carlos.santanadeoliveira@canonical.com](mailto:carlos.santanadeoliveira@canonical.com))*
+
+Cloud-init is an industry-standard multi-distribution method for cross-platform cloud instance initialisation.
+Ubuntu WSL users can now leverage it to perform an automatic setup to get a working instance with minimal touch.
+That feature is currently experimental and, thus only available on the UbuntuPreview app. Soon that will be available for
+the latest LTS application as well.
+
+## What you will learn:
+
+- How to write cloud-config user data to a specific WSL instance.
+- How to automatically set up a WSL instance with cloud-init.
+- How to verify that cloud-init succeeded with the configuration supplied.
+
+## What you will need:
+
+- Windows 11 with WSL 2 already enabled
+- The latest UbuntuPreview application from Microsoft Store.
+
+## Write the cloud-config file
+
+Locate your Windows user home directory. It typically is `C:\Users\<YOUR_USER_NAME>`.
+
+> You can be sure about that path by running `echo $env:USERPROFILE` in PowerShell.
+
+Inside your Windows user home directory, create a new folder named `.cloud-init` (notice the `.` à la Linux
+configuration directories), and inside the new directory, create an empty file named `Ubuntu-Preview.user-data`. That file name must
+match the name of the distro instance that will be created in the next step.
+
+Open that file with your text editor of choice (`notepad.exe` is just fine) and paste in the following contents:
+
+```yaml
+#cloud-config
+locale: pt_BR
+users:
+- name: jdoe
+  gecos: John Doe
+  groups: [adm,dialout,cdrom,floppy,sudo,audio,dip,video,plugdev,netdev]
+  sudo: ALL=(ALL) NOPASSWD:ALL
+  shell: /bin/bash
+
+write_files:
+- path: /etc/wsl.conf
+  append: true
+  content: |
+    [user]
+    default=jdoe
+
+packages: [ginac-tools, octave]
+
+runcmd:
+   - git clone https://github.com/Microsoft/vcpkg.git /opt/vcpkg
+   - /opt/vcpkg/bootstrap-vcpkg.sh
+```
+
+Save it and close it.
+
+> That example will create a user named `jdoe` and set it as default via `/etc/wsl.conf`, install the packages
+> `ginac-tools` and `octave` and install `vcpkg` from the git repository, since there is no deb or snap of that
+> application (hence the reason for being included in this tutorial - it requires an unusual setup).
+
+## Register a new Ubuntu-Preview instance
+
+In PowerShell, run:
+
+```powershell
+ubuntupreview.exe install --root
+```
+
+We skip the user creation since we expect cloud-init to do it.
+
+> If you want to be sure that there is now an Ubuntu-Preview instance, run `wsl -l -v`.
+> Notice that the application is named `UbuntuPreview` but the WSL instance created is named `Ubuntu-Preview`.
+> See more about that naming convention in [our reference documentation](../reference/distributions.md#naming).
```

Comment:
From the RTD builder:
```
/home/docs/checkouts/readthedocs.org/user_builds/canonical-ubuntu-wsl/checkouts/cloud-init-tut/docs/tutorials/cloud-init.md:72: WARNING: local id not found in doc 'reference/distributions': 'naming' [myst.xref_missing]
```

I think the `.md` file extension should not be there. Alternatively, you can add a tag in the destination https://docs.readthedocs.io/en/stable/guides/cross-referencing-with-sphinx.html

Reply:
That's interesting. Without the `.md` the local website fails to navigate. But both with or without `.md` I'm noticing in the website build process that reference not found error. Only referencing with the explicit target prevented that failure.